docs/src/guide/installation.md

```diff
@@ -32,145 +32,278 @@ Please help us [keep these instructions up-to-date][contribute]!
-	pip3 install modm gdbgui
+```sh
+sudo apt install python3 python3-pip scons git
+pip3 install modm
+```
+
+We recommend using a graphical frontend for GDB called [gdbgui][]:
+
+```sh
+pip3 install gdbgui
+```
 
 !!! info "Python Packages in PATH"
 	The pip3 command installs executables into `~/.local/bin`, which
 	must be added to PATH if it is not already the case.
 	Add the following line to the end of your `~/.bashrc` file:
 
-		export PATH="~/.local/bin:$PATH"
+	```sh
+	export PATH="~/.local/bin:$PATH"
+	```
+
+!!! warning "Ubuntu ≤ 19.10"
+	Ubuntu ≤ 19.10 unfortunately defaults to Python 2, however, our SCons tools
+	require Python 3. For a less intrusive way to run all scons scripts with
+	Python 3 add this to your `.bashrc` or `.zshrc`:
+
+	```sh
+	alias scons="/usr/bin/env python3 $(which scons)"
+	```
+
+	If you get errors about missing `pyelftools`, check that you're really
+	running SCons with Python 3!
+
+We use [Doxypress][doxypress_binaries] to generate the API documentation:
+
+```sh
+sudo mkdir /opt/doxypress
+wget -O- https://download.copperspice.com/doxypress/binary/doxypress-1.4.1-ubuntu20.04-x64.tar.bz2 | sudo tar xj -C /opt/doxypress
+```
+
+Add the directory to your `PATH` variable in `~/.bashrc`:
+
+```sh
+export PATH="/opt/doxypress:$PATH"
+```
 
-For host system (x86_64/arm64) targets:
 
-	sudo apt install gcc build-essential libboost-all-dev
+#### ARM Cortex-M
 
-For ARM Cortex-M targets:
+Install the [pre-built ARM toolchain][gcc-arm-toolchain]:
 
-	sudo apt install openocd
+```sh
+wget -O- https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2 | sudo tar xj -C /opt/
+```
+
+Add it to your `PATH` variable in `~/.bashrc`:
+
+```sh
+export PATH="/opt/gcc-arm-none-eabi-10-2020-q4-major/bin:$PATH"
+```
 
 !!! warning "Ubuntus 'gcc-arm-none-eabi' package"
-	The Ubuntu package 'gcc-arm-none-eabi' causes [issues at the moment](https://github.com/modm-io/modm/issues/468) and ships only version 9.3.
-	We recommend using the toolchain provided by ARM:
+	The Ubuntu package 'gcc-arm-none-eabi' causes [issues at the moment][armgcc-issues]
+	and ships only version 9.3.
 
-		wget -O- https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2 | sudo tar xj -C /opt/
-	
-	Add it to your systems PATH variable by appending the following line to `~/.bashrc`:
+Install the OpenOCD tool:
 
-		export PATH="/opt/gcc-arm-none-eabi-10-2020-q4-major/bin:$PATH"
+```sh
+sudo apt install openocd
+```
 
-For Atmel/Microchip AVR targets:
+!!! warning "OpenOCD < v0.11.0"
+	Make sure to get at least OpenOCD release v0.11, since v0.10 is too old for
+	some targets (STM32G0, STM32G4, STM32F7). You may need to get the
+	[HEAD version of OpenOCD from source][openocd-source], install it and add it
+	to the beginning of your `$PATH`. [Here is a detailed guide][openocd-compile].
```

Comment:
... or use pre-compiled packages from my github https://github.com/rleh/openocd-build/releases?
(RPM for Fedora/CentOS/.../RHEL works great; Debian package installs into `/opt/openocd` and is undocumented...)

Reply:
Absolutely, but I don't know how the commands for that are? Please push a fixup commit with the changes.

Reply:
I'm going to put install instructions into the openocd-build repos README and would then link to it. Ok?

---

I haven't noticed OpenOCD has finally released a new version (0.11) http://openocd.org/2021/03/openocd-0-11-0-released/ 😱😱😱😱
Nice!

Reply:
I added the link.